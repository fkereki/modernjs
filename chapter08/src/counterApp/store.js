/* @flow */

import { createStore } from "redux";

import { reducer } from "./counter.reducer.js";

export const store = createStore(reducer);
