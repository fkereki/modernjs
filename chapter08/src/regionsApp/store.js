/* @flow */

import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";

import { reducer } from "./world.reducer";

export const store = createStore(reducer, applyMiddleware(thunk));
