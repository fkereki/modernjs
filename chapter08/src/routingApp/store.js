/* @flow */

import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";

import { reducer } from "./login.reducer.js";

export const store = createStore(reducer, applyMiddleware(thunk));
