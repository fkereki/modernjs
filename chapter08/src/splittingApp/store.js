/* @flow */

import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";

import { reducer } from "./login.reducer";

export const store = createStore(reducer, applyMiddleware(thunk));
