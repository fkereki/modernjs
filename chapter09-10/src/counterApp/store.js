/* @flow */

import { createStore, applyMiddleware } from "redux";
import { createLogger } from "redux-logger";
import { composeWithDevTools } from "redux-devtools-extension";

import { reducer } from "./counter.reducer.js";

const logger = createLogger({
    diff: true,
    duration: true
});

// The following uses only redux-logger
/*
export const store =
    process.env.NODE_ENV === "development"
        ? createStore(reducer, applyMiddleware(logger))
        : createStore(reducer);
*/

// The following uses redux-devtools-extension in addition to redux-logger

export const store = createStore(
    reducer,
    composeWithDevTools(applyMiddleware(logger))
);
