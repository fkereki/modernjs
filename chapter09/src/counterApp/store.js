/* @flow */

import { createStore, applyMiddleware } from "redux";
import { createLogger } from "redux-logger";

import { reducer } from "./counter.reducer.js";

const logger = createLogger({ diff: true, duration: true });

export const store = createStore(reducer, applyMiddleware(logger));
