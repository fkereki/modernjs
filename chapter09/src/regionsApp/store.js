/* @flow */

import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { createLogger } from "redux-logger";
import { composeWithDevTools } from "redux-devtools-extension";

import { reducer } from "./world.reducer.js";

const logger = createLogger({ duration: true });

// The following uses redux-logger, only.
/*
export const store = createStore(reducer, applyMiddleware(thunk, logger));
*/

// The following uses redux-devtools-extension in addition to redux-logger

export const store = createStore(
    reducer,
    composeWithDevTools(applyMiddleware(thunk, logger))
);
