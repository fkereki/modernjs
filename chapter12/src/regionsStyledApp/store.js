/* @flow */

import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "remote-redux-devtools-sp";

import { reducer } from "./world.reducer";

export const store = createStore(
    reducer,
    composeWithDevTools(applyMiddleware(thunk))
);
