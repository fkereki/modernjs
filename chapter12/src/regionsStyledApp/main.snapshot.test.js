/* @flow */

import React from "react";
import { Provider } from "react-redux";
import TestRenderer from "react-test-renderer";

import { Main } from "./main.component";
import { store } from "./store";

const fakeDeviceData = {
    isTablet: false,
    isPortrait: true,
    height: 1000,
    width: 720,
    scale: 1,
    fontScale: 1
};

describe("Main component", () => {
    it("renders in portrait mode", () => {
        const tree = TestRenderer.create(
            <Provider store={store}>
                <Main
                    deviceData={{ ...fakeDeviceData, isPortrait: true }}
                />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders in landscape mode", () => {
        const tree = TestRenderer.create(
            <Provider store={store}>
                <Main
                    deviceData={{ ...fakeDeviceData, isPortrait: false }}
                />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
