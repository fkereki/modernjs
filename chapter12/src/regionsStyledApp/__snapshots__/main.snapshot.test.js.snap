// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renders in landscape mode 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    hidden={true}
    onLayout={[Function]}
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Loading countries...
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          No regions.
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Main component renders in portrait mode 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    hidden={true}
    onLayout={[Function]}
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Loading countries...
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          No regions.
        </Text>
      </View>
    </View>
  </View>
</View>
`;
